src/analyzeCode.ts

```diff
@@ -25,32 +30,149 @@ export function analyzeCode(code: string): VariableStatementAnalysis[] {
+                variableValue: processExpression(
+                    node.initializer,
+                    detectedVariableStatements
+                ),
             });
+        } else if (ts.isExpressionStatement(node)) {
+            console.log(detectedVariableStatements);
+            const nodeExpression = node.expression;
+            if (ts.isBinaryExpression(nodeExpression)) {
+                const elementIndex = detectedVariableStatements.findIndex(
+                    (variables) => {
+                        return (
+                            variables.variableName ===
+                            nodeExpression.left.getText()
+                        );
+                    }
+                );
+                //since right would always be binary expression we want to process that and update the variable value
+                detectedVariableStatements[elementIndex].variableValue =
+                    processExpression(
+                        nodeExpression.right,
+                        detectedVariableStatements
+                    );
+            }
         }
     }
-
     // iterate through source file searching for variable statements
     visitNodeRecursive(sourceFile, visitVariableStatement);
-    // sourceFile.statements.forEach((node) => {
-    //     visitNodeRecursive(node, visitVariableStatement);
-    // });
 
     return detectedVariableStatements;
 }
 
+//TODO: can we merge from a record to something else that supports boolean 
+const Operations: Record<number, (a: number, b: number) => number> = {
+    39: (a: number, b: number): number => a + b,
+    40: (a: number, b: number): number => a - b,
+    41: (a: number, b: number): number => a * b,
+    42: (a: number, b: number): number => a ** b,
+    43: (a: number, b: number): number => a / b,
+    44: (a: number, b: number): number => a % b,
+
+    //36: (a:number, b:number): boolean => a === b
+    //37: (a:number, b:number): boolean => a !== b
+};
+
+function getValue(
+    node: ts.NumericLiteral | ts.Identifier,
+    detectedVariableStatements: VariableStatementAnalysis[]
+): number {
+    if (ts.isNumericLiteral(node)) {
+        return parseFloat(node.getText());
+    } else {
+        const identifiervalue = detectedVariableStatements.find((variables) => {
+            return variables.variableName === node.getText();
+        });
+        if (identifiervalue === undefined) {
+            throw new Error(
+                "Identifier cannot be found or undefined, please define a variable before using it"
+            );
+        }
+
+        return identifiervalue.variableValue;
+    }
+}
+
+function processExpression(
+    node: ts.Expression | ts.NumericLiteral | undefined,
+    detectedVariableStatements: VariableStatementAnalysis[]
+): number {
+    if (node === undefined) {
+        throw new Error("Expression is undefined");
+    }
+
+    if (ts.isBinaryExpression(node)) {
+        if (
```

Comment:
I have cleaned up the code with recursion maybe give it a second look
but regarding the number | undefined... that will be problematic for the Record that maps to functions because you can't do undefined + 2 or something.... any solution for this?

Reply:
Do you know a way to throw an error and return undefined at the same time?

Reply:
Function now returns undefined